modules/planning/navi/common/local_path.h

```diff
@@ -0,0 +1,107 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ * @brief This file provides the declaration of the class "LocalPath".
+ */
+
+#ifndef MODULES_PLANNING_NAVI_LOCAL_PATH_H_
+#define MODULES_PLANNING_NAVI_LOCAL_PATH_H_
+
+#include <memory>
+#include <vector>
+
+#include "modules/common/math/vec2d.h"
+#include "modules/common/proto/pnc_point.pb.h"
+
+/**
+ * @namespace apollo::planning
+ * @brief apollo::planning
+ */
+namespace apollo {
+namespace planning {
+/**
+ * @class LocalPath
+ * @brief LocalPath is used to privide some path point operations for decider
+ */
+class LocalPath {
+ public:
+  LocalPath(std::vector<common::PathPoint> &path_points)
+      : path_points_(path_points) {
+    points_.clear();
+    for (int i = 0; i < path_points_.size(); ++i) {
+      points_.emplace_back(path_points_[i].x(), path_points_[i].y());
+    }
+  }
+
+  virtual ~LocalPath() = default;
+
+  /**
+   * @brief get the first y of points vector.
+   * @return std::shared_ptr<double> if result is not valid return nullptr.
+   */
+  std::shared_ptr<double> GetInitY();
+
+  /**
+   * @brief get the xy coordinate vector.
+   * @return std::vector<Vec2d> the xy coordinate vector.
+   */
+  std::vector<Vec2d> GetXYPoints();
+
+  /**
+   * @brief get points range.
+   * @return int the valid points range.
+   */
+  int GetRange();
+
+  /**
+   * @brief shift the points.
+   * @param dist shift distance.
+   * @return void.
+   */
+  void Shift(const double dist);
+
+  /**
+   * @brief cut the points.
+   * @param dist cut distance.
+   * @return void.
+   */
+  void Cut(const double dist);
+
+  /**
+   * @brief resample the points.
+   * @return void.
+   */
+  void Resample();
+
+  /**
+   * @brief merge the points with local path.
+   * @param common::Path local_path the path which is used to merge with points.
+   * @param weight the weight coefficient.
+   * @return void.
+   */
+  void Merge(const common::Path local_path, const double weight);
+
+ private:
+  std::vector<common::PathPoint> path_points_;
+  std::vector<Vec2d> points_;
```

Comment:
Please double check other files for putting "using" in header files. If ```std::vector<Vec2d>``` is compilable, it indicates an contamination.